lib/pages/conversation_page.dart

```diff
@@ -0,0 +1,158 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/scheduler.dart';
+import 'package:lend_a_hand/services/auth_service.dart';
+
+class Conversation extends StatefulWidget {
+  final FirebaseUser firebaseUser;
+  final String reciever;
+
+  const Conversation({Key key, this.firebaseUser, this.reciever}) : super(key: key);
+
+  @override
+  _ConversationState createState() => _ConversationState();
+}
+
+class _ConversationState extends State<Conversation> {
+  final FirebaseAuth _auth = FirebaseAuth.instance;
+  final Firestore _firestore = Firestore.instance;
+  var _messages;
+
+  TextEditingController _controller = TextEditingController();
+  ScrollController _scrollController = ScrollController();
+
+  Future<void> callback() async {
+    if (_controller.text.length > 0) {
+      var data = {'text': _controller.text, 'sender': widget.firebaseUser.email, 'date': DateTime.now().toIso8601String().toString()};
+      await _firestore
+          .collection('chats')
+          .document(widget.firebaseUser.email)
+          .updateData({'messages': FieldValue.arrayUnion([data])});
+      _controller.clear();
+      _scrollController.animateTo(_scrollController.position.maxScrollExtent,
+          curve: Curves.easeOut, duration: Duration(milliseconds: 300));
+    }
+  }
+
+  @override
+  void initState() {
+    super.initState();
+
+    _messages = StreamBuilder<DocumentSnapshot>(
+      stream: _firestore.collection('chats')
+          .document(widget.firebaseUser.email)
+          .snapshots(),
+      builder: (context, snapshot) {
+        if (!snapshot.hasData)
+          return Center(child: CircularProgressIndicator());
+
+        SchedulerBinding.instance.addPostFrameCallback((_) {
+          _scrollController.animateTo(
+            _scrollController.position.maxScrollExtent,
+            curve: Curves.easeOut,
+            duration: const Duration(milliseconds: 300),
+          );
+        });
+
+        DocumentSnapshot doc = snapshot.data;
```

Comment:
You can remove this local variable and in the line below add `snapshot.data`. Or better yet, save the location rather than the whole `Snapshot`